test/sol5/kyberNetworkProxy.js

```diff
@@ -578,6 +578,41 @@ contract('KyberNetworkProxy', function(accounts) {
             });
         }
     });
+    
+    describe("test enable", async () => {
+        it("test set enable with admin", async () => {
+            await network.setEnable(false, { from: admin });
+            let enable = await networkProxy.enabled();
+            assert(!enable);
+            await network.setEnable(true, { from: admin });
+            enable = await networkProxy.enabled();
+            assert(enable);
+        });
+    });
+
+    describe("test swap function (backward compatible)", async () => {
+        it("swapEtherToToken", async () => {
+            let initialTakerBalance = await tokens[1].balanceOf(taker);
+            let tokenAdd = tokens[1].address;
+            let srcQty = (new BN(2)).mul((new BN(10)).pow(ethDecimals));
+            let proxyRate = await networkProxy.getExpectedRate(ethAddress, tokenAdd, srcQty);
+            Helper.assertEqual(proxyRate.worstRate, proxyRate.expectedRate.mul(new BN(97)).div(new BN(100)));
+            await networkProxy.swapEtherToToken(tokenAdd, proxyRate.expectedRate, { from: taker, value: srcQty });
+            let dstQty = Helper.calcDstQty(srcQty, ethDecimals, tokenDecimals[1], proxyRate.expectedRate);
+            let expectedBalance = new BN(initialTakerBalance).add(dstQty);
+            await Helper.assertSameTokenBalance(taker, tokens[1], expectedBalance);
+        });
+
+        it("swapTokenToEther", async () => {
+            let srcToken = tokens[1]
+            let tokenAdd = tokens[1].address;
+            let srcQty = (new BN(5)).mul((new BN(10)).pow(tokenDecimals[1]));
+            await srcToken.approve(networkProxy.address, srcQty, { from: taker });
+            let proxyRate = await networkProxy.getExpectedRate(tokenAdd, ethAddress, srcQty);
```

Comment:
also here.
Don't use the backward compatible getRate API

our orientation should be using the new get rate API